### PR TITLE
Remove .btn-block for utilities

### LINKED ISSUE #30223: Add support for block buttons based on specific screen widths, eg. btn-xs-block
This is an easy feature that anyone can implement with a little extra css, but I think it is a extremely common case that I think it would be necessary and useful for it to be built into Bootstrap.

### PR DESCRIPTION
**Pull request by mdo**
Alternate fix for #30376, which closes #30223.

### CODE REVIEW

File: site/content/docs/4.3/migration.md [hunk 1]
```diff
@@ -129,6 +129,10 @@ Badges were overhauled to better differentiate themselves from buttons and to be
 - **Todo:** Removed `.badge-pill` for the `.rounded-pill` utility class
 - **Todo:** Removed badge's hover and focus styles for `a.badge` and `button.badge`.
 
+### Buttons
+
+- Removed `.btn-block` as it duplicated functionality of utilities, but in a more limited way. With a utilities based approach, "block buttons" are now easily made responsive.
```
Comment (XhmikosR) [inline site/content/docs/4.3/migration.md:134]: I think we should mention the exact replacement classes to make things easier for people 🙂

**Comment by MartijnCuppens** [comment]
I don't think we need the `.d-block` class, only `.w-100` should be ok. We can even ditch the `mb-2`, since the inline blocks already have some spacing.

**Comment by mdo** [comment]
Might not go this route after all actually... requires more utility classes to _undo_ the responsive blocks (assuming block on mobile, not block on say medium). 🤔

**Comment by mdo** [comment]
Closing per my last comment.

**Pull request closed without merging**